tss-esapi/src/traits.rs

```diff
@@ -15,4 +20,74 @@ pub trait Marshall: Sized {
 pub trait UnMarshall: Sized {
     /// Creates the type from marshalled data.
     fn unmarshall(marshalled_data: &[u8]) -> Result<Self>;
+
+    /// Creates the type from the marshalled data, and modifies
+    /// the `offset` to point to the first byte in the `marshalled_data`
+    /// buffer which was not used in the conversion.
+    fn unmarshall_offset(_marshalled_data: &[u8], _offset: &mut u64) -> Result<Self> {
+        unimplemented!();
```

Comment:
Unimplemented will panic. I wonder what's the reason for not leaving it empty (with no implementation)? Because other types are not implementing that yet?

At the end of the day I guess it would be good to provide default implementation to the `unmarshall` that just calls `unmarshall_offset(marshalled_data, 0)` :thinking:

Reply:
Okay, this sounds fair. Thanks for taking the time to explain!